packages/python/plotly/plotly/express/_core.py

```diff
@@ -813,6 +813,12 @@ def build_dataframe(args, attrables, array_attrables):
     array_attrables : list
         argument names corresponding to iterables, such as `hover_data`, ...
     """
+    args = dict(input_args)
+    for field in args:
+        if field in array_attrables and isinstance(
+            args[field], pd.core.indexes.base.Index
+        ):
+            args[field] = list(args[field])
```

Comment:
following from comment above... if we just always do this, instead of in an `if` then I think we're safe (for now!)

Reply:
good idea! Then we don't have to do a deepcopy of the whole `args`, it's better not to copy the `dataframe` in particular.